insitu/models.py

```diff
@@ -179,71 +124,60 @@ class LoggedAction(models.Model):
     target_note = models.TextField(blank=True)
 
 
-class ValidationWorkflowModel(WorkflowEnabled, models.Model):
-    state = StateField(ValidationWorkflow)
+def check_owner_user(instance, user):
+    """
+    Check if the user is the creator or if the is user is in the creator's
+    team
+    """
+    return (
+        user == instance.created_by
+        or user in instance.created_by.team.teammates.all()
+        or instance.has_user_perm(user)
```

Comment:
shouldn't this be something like
`
instance.has_user_perm(user) AND ( user == instance.created_by  or user in instance.created_by.team.teammates.all() )
`
At the moment it looks like  if the user has delete/change permissions on the object he will pass this check even if  (s)he is not the owner or teammate.